### DISCUSSION TITLE
2021-11-11 weekly planning

### DISCUSSION THREAD

**Discussion by kgodey** [mathesar-foundation/mathesar#820] [Maintainer discussions]
## About
Each week (usually on Tuesday), we start a discussion covering Mathesar's progress, team member check ins, and general updates. Everyone is welcome to add a new topic to discuss or to comment on existing topics.

## Discussion topics
- [Progress on [06] General Improvements #4](https://github.com/centerofci/mathesar/discussions/820#discussioncomment-1627919)
- [Progress on [07] Working with Tables](https://github.com/centerofci/mathesar/discussions/820#discussioncomment-1627925)
- [Progress on [08] Initial Data Types](https://github.com/centerofci/mathesar/discussions/820#discussioncomment-1627930)
- [Progress on [09] Working with Views](https://github.com/centerofci/mathesar/discussions/820#discussioncomment-1627977)
- [New Team Members and Contributors](https://github.com/centerofci/mathesar/discussions/820#discussioncomment-1628012)
- [Comms Assignee schedule](https://github.com/centerofci/mathesar/discussions/820#discussioncomment-1628019)
- [Kriti's check in](https://github.com/centerofci/mathesar/discussions/820#discussioncomment-1628034)
- [Mukesh's check in](https://github.com/centerofci/mathesar/discussions/820#discussioncomment-1630128)
- [Ghislaine's check in](https://github.com/centerofci/mathesar/discussions/820#discussioncomment-1631845)
- [Pavish's check in](https://github.com/centerofci/mathesar/discussions/820#discussioncomment-1632192)
- [Sean's check in](https://github.com/centerofci/mathesar/discussions/820#discussioncomment-1632377)
- [Brent's check in](https://github.com/centerofci/mathesar/discussions/820#discussioncomment-1633309)

## Weekly check-in
@mathemancer, @ghislaineguerin, @pavish, @seancolsen, @silentninja please respond to the weekly check-in as a top-level response and edit the discussion topics above to add your check in as a topic. Others are also welcome to respond.

Based on feedback from last week, I'm combining some questions this week to hopefully make the check-in a bit quicker (but still useful).

- How did the last week go? What went well, what could have gone better?
- What are you planning to work on over the next week? Do you anticipate any blockers?
- Anything else you'd like to add?

**Comment by kgodey** [comment]
## Progress on [[06] General Improvements #4](https://github.com/centerofci/mathesar/milestone/31)
We're almost done! Out of 5 issues remaining last week, 3 are complete, 1 is in review, and 1 is close to being done. @pavish and @seancolsen, please prioritize getting the last two merged:
- #639 
- #651

**Comment by kgodey** [comment]
## Progress on [[07] Working with Tables](https://github.com/centerofci/mathesar/milestone/8)
We've also made progress on this milestone last week with a couple of issues merged. We have 13 open tickets (excluding meta issues), some of which are in progress already. I'm only covering the ones that are not yet assigned or started.

- @ghislaineguerin please prioritize this design issue once you're done with the Foreign Key spec.
    - #780 
- @seancolsen @pavish @mr-gabe49 I don't think we can complete the milestone this week, but hopefully we can make a dent in it. Please pick up these unclaimed issues if you can (and it won't cause too many merge conflicts with other issues that are being worked on).
    - #445 
    - #531 
    - #770 
    - #773
    - #775  
    - #631

**Comment by kgodey** [comment]
## Progress on [[08] Initial Data Types](https://github.com/centerofci/mathesar/milestone/11)

Our priority is to finish the backend work for this milestone as soon as possible. There are a few buckets of work here:

- Improving our support for columns, #658. @silentninja, I assume you'll be done with that this week?
- Underlying data type support
  - @eito-fis, will you be able to finish up #426 this week? 
  - @mathemancer, #387 only needs tests, IIRC. Could you handle that this week?
  - The remaining types (`MONEY`, `INTERVAL`, and `TIMESTAMP` are currently blocked by the other tickets getting done first.
- Custom grouping options for data types
  - @mathemancer, will you be able to finish the grouping refactor this week? Once that's done, we should be able to unblock the others.
- Custom filtering options for data types.
   - There are some unclaimed tickets here, @silentninja @mathemancer, please feel free to take them if you have the time.
      - #410 
      - #385
      - #413 
      - #406
- Custom display types work
   - Currently blocked by column work. 
- #648 - @mathemancer, what is the status of this?

**Reply by silentninja** [reply]
> * Improving our support for columns, [Set up `Column` Django model with support for display options #658](https://github.com/centerofci/mathesar/issues/658). @silentninja, I assume you'll be done with that this week?

Yes

**Reply by eito-fis** [reply]
> @eito-fis, will you be able to finish up Handle TIME data type in the backend #426 this week?

I have a break next week and was hoping to get it done then, I don't think I can do it this week.

**Reply by kgodey** [reply]
@eito-fis I think next week should be fine. Please let us know if you cannot do it by end of next week, someone else might need to finish it up since it's starting to block other work. Thanks for your work on it so far!

**Reply by mathemancer** [reply]
For the grouping refactor, I can be done quite soon, subject to agreement that we should defer fixing and unifying the way we pass records request arguments around until later (that kind of got tangled up in the grouping, since the addition of different grouping types, with different argument combos to define their parameters gets pretty wild just being passed about as strings).

I also thing the `BOOLEAN` implementation should be quick and is mostly done.  There are even a bunch of tests already.  We just need to verify that the tests are defining the correct intended behavior (I made some decisions, as you recall, about what we should/shouldn't call boolean way back when).  I'll double-check against the current issue.

**Reply by kgodey** [reply]
@mathemancer Yes, that's fine, I think we should defer it. Please create an issue for it if needed.

**Comment by kgodey** [comment]
## Progress on [[09] Working with Views](https://github.com/centerofci/mathesar/milestone/13)

- Product work on defining Views is in progress and will hopefully be done this week, please see #442 for the meta-issue tracking those.
- We're restarting design work on Views based on the new issues that have been reviewed by the team, starting with #466.
- There are no action items for anyone except me yet. Please keep an eye out for new issues to review as I complete them.

**Comment by kgodey** [comment]
## New Team Members and Contributors

I'd like to introduce a couple of [team members](https://wiki.mathesar.org/en/team):
- @silentninja (Mukesh) is a new full-time member of the Mathesar team. He'll be working on the backend.
- @ppii775 (Han) is a member of our community team working on design. 

Please tell us a little bit about yourselves in this thread.

Mathesar has also had a few new contributors in the last few weeks. Thank you @gagan2005, @fkarg, @HenryRocha, @VFermat, @rajeshkumarrs, and I'm sorry if I missed anyone.

**Reply by ppii775** [reply]
Hi Everyone! Excited to join the team : )
I'm Han, a graduate student at UC Berkeley. I'm a product designer and started working on Mathesar for a 4-week class design project. I decided to continue contributing to the product and learning more from everybody on the team : ) I'm planning to work 5-10hrs a week for now, depending on school schedules.

**Reply by mathemancer** [reply]
We're glad to have you onboard!

**Comment by kgodey** [comment]
## Comms Assignee schedule

Upcoming Comms Assignee schedule:
| Week | Person |
|-|-|
| Nov 8 to Nov 12 | Ghislaine |
| Nov 15 to Nov 19 | Brent |
| Nov 22 to Nov 26 | Kriti |
| Nov 29 to Dec 3 | Pavish |
| Dec 6 to Dec 10 | Mukesh |
| Dec 13 to Dec 17 | Sean |

You can also find the schedule and instructions for the Comms Assignee on the [wiki page](https://wiki.mathesar.org/team/guide/comms-assignee). Please feel free to update the wiki page if you think of anything useful during your shift.

**Comment by kgodey** [comment]
## Kriti's check in

>  How did the last week go? What went well, what could have gone better?

I think it went all right. We merged in quite a few pull requests and I'm glad that we completed some features. It's also great to have new team members. I would have liked to make more progress on Views product work.

> What are you planning to work on over the next week? Do you anticipate any blockers?

Other than general planning and follow up,
- Views product work.
- #560 
- Budget and board meeting prep.

**Comment by silentninja** [comment]
## Mukesh's check in

>  How did the last week go? What went well?

It was exciting. I was able to complete some pending PR, have interesting discussions, meet up with teammates and have a fascinating conversation and was able to get to know them better. 

> what could have gone better?

Goofed up in a few places, could have been better if I read the wiki. Could have been better if I was able to complete the column model issue got delayed to personal issues.

> What are you planning to work on over the next week?

* Planning to get the column model PR approved #815 
* Work on the filtering and display options issues.
* Add my observations to wiki  

> Do you anticipate any blockers?

None

**Reply by kgodey** [reply]
It sounds like you've had a productive week. :)

**Comment by ghislaineguerin** [comment]
## Ghislaine's Check-in

> How did the last week go? What went well, what could have gone better?

It was good, in terms of progress with the current design work. However, I felt that the scope was increasing and that was a bit stressful. I now see that this is also a consequence of the design discussions and that this additional work would have been done later anyway.

> What are you planning to work on over the next week? Do you anticipate any blockers?

I plan on working on addressing review comments for Foreign key specs and design issues that were open during/after implementation.

> Anything else you'd like to add?

Nothing for now.

**Comment by pavish** [comment]
## Pavish's Check-in

> How did the last week go? What went well, what could have gone better?

It went okay. I reviewed and merged in some good feature PRs, and got the PR for url sync ready. I appreciate the discussions we've been having from issue descriptions to UX. I got stuck at the integ test env setup again which delayed things a bit. It should be ready soon.

> What are you planning to work on over the next week? Do you anticipate any blockers?

I'm prioritizing on getting the open PRs merged first. After that, I intend to pick some of the self assigned issues from [07] Working with Tables.

> Anything else you'd like to add?

Nothing at the moment.

**Reply by kgodey** [reply]
Is there anything any of us can do to help with integration testing? Sometimes it's also good to take a break to work on something else for a bit if you're stuck.

**Reply by pavish** [reply]
>  it's also good to take a break to work on something else for a bit

I took a break to work on the url sync which is now merged. I've also started on the type abstraction required for setting default value of a column.

**Comment by seancolsen** [comment]
## Sean's check in

How did the last week go? What went well, what could have gone better?

- Pretty well. Happy to be getting some more PRs closed.
- Also happy to be continuing to get a better sense of our collective product vision through all these UX discussions.

What are you planning to work on over the next week? Do you anticipate any blockers?

- Aiming to work on areas of the code that will minimize conflict with Pavish's work. Currently that's the toast system. Will touch base as needed to plan my work accordingly. I'm fairly motivated to work also on the modal system, the confirmation system, and the dropdown system too, because I've been seeing new code that could be much simpler if we could flesh out those low-level systems first. But I recognize that work is not part of our current milestone so I'll need to balance priorities.

Anything else you'd like to add?

- I've probably mentioned to all of you how difficult my wife's back injury has been. It's been terrible for her and has also made like outside of work incredibly hard for me because I barely have time to eat. After almost three months, and she's finally starting to improve. In the past few days she's been able to pick up our kids a few times (one at a time) which is amazing. She's been able to spend some time taking care of the kids on her own. It's so reliving to all of us. She's still taking it very slowly, but we're hopeful that the worst of it might be over.

**Reply by pavish** [reply]
I'm really glad for both of you, I hope things improve a lot more soon :)


> I'm fairly motivated to work also on the modal system, the confirmation system, and the dropdown system too.

I think we should work on them soon since they will make it easier to work on feature issues.

Regarding the dropdown system, I believe you're talking about the dropdown-menu. I think Menu should be a separate low level component of it's own and we can slot it within Dropdown whenever required. For eg., in the column header, we have a dropdown menu, which changes it's structure entirely when we try to switch column type. Having a separate Menu component would give us more flexibility.

**Reply by kgodey** [reply]
I'm very glad to hear that your wife is doing better. :)

**Comment by mathemancer** [comment]
> How did the last week go? What went well, what could have gone better?

- Went well:  Making progress on the grouping API changes.  They were more extensive than I originally wanted, but they're working.  The changes should not only add the feature needed, but generally improve query performance (especially when groups are requested).  
- Could've gone better:  Still underwhelming velocity.

> What are you planning to work on over the next week? Do you anticipate any blockers?

I'm still working on the Grouping API.  After that, I want to get into more type work.  The Grouping API changes are breaking, which will introduce a dependency with front end.  We'll handle it as discussed in other places.

> Anything else you'd like to add?

Not at this moment.
